framework/src/media/audio/audio_manager.c

```diff
@@ -725,21 +731,23 @@ audio_manager_result_t set_audio_stream_in(unsigned int channels, unsigned int s
 		goto error_with_pcm;
 	}
 
+	card->resample.format = pcm_format_to_bits(pcm_get_format(card->pcm)) >> 3;	// ToDo: Change into the argument "format".
 	if (sample_rate == config.rate) {
 		card->resample.necessary = false;
 	} else {
 		card->resample.necessary = true;
-		card->resample.from = config.rate;
-		card->resample.to = sample_rate;
+		card->resample.sample_src = config.rate;
+		card->resample.sample_dst = sample_rate;
 		card->resample.ratio = (float)sample_rate / (float)config.rate;
 		card->resample.handle = src_init(CONFIG_AUDIO_RESAMPLER_BUFSIZE);
-		card->resample.buffer = malloc((int)((float)get_input_frames_to_byte(get_input_frame_count()) / card->resample.ratio) + 1);	// +1 for floating point margin
+		card->resample.buffer = malloc((int)((float)get_user_input_frames_to_byte(get_input_frame_count()) / card->resample.ratio) + 1);	// +1 for floating point margin
```

Comment:
would you please test 'ceil' instead of this?

Reply:
Ok, I'll modify it after basic functions are implemented.